polyglot/piranha/src/cleanup_rules/go/rules.toml

```diff
@@ -0,0 +1,487 @@
+# License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+# The language specific rules in this file are applied after the API specific change has been performed.
+
+# Dummy rule that acts as a junction for all boolean based cleanups
+[[rules]]
+name = "boolean_literal_cleanup"
+
+# Before :
+#  !false
+# After :
+#  true
+#
+[[rules]]
+groups = ["boolean_expression_simplify"]
+name = "simplify_not_false"
+query = """
+(
+    (unary_expression
+        operator: "!"
+        operand: [
+            (false)
+            (parenthesized_expression (false))
+        ]
+    ) @unary_expression
+)
+"""
+replace = "true"
+replace_node = "unary_expression"
+
+# Before :
+#  !true
+# After :
+#  false
+#
+[[rules]]
+groups = ["boolean_expression_simplify"]
+name = "simplify_not_true"
+query = """
+(
+    (unary_expression
+        operator: "!"
+        operand: [
+            (true)
+            (parenthesized_expression (true))
+        ]
+    ) @unary_expression
+)
+"""
+replace = "false"
+replace_node = "unary_expression"
+
+[[rules]]
+groups = ["boolean_expression_simplify"]
+name = "simplify_true_and_something"
+query = """
+(
+    (binary_expression
+        left: [(true) (parenthesized_expression (true))]
+        operator:"&&"
+        right : (_) @rhs
+    ) @binary_expression
+)
+"""
+replace = "@rhs"
+replace_node = "binary_expression"
+
+# Before :
+#  abc() && true
+# After :
+#  abc()
+#
+[[rules]]
+groups = ["boolean_expression_simplify"]
+name = "simplify_something_and_true"
+query = """
+(
+    (binary_expression
+        left : (_) @lhs
+        operator:"&&"
+        right: [(true) (parenthesized_expression (true))]
+    ) @binary_expression
+)
+"""
+replace = "@lhs"
+replace_node = "binary_expression"
+
+# Before :
+#  false && abc()
+# After :
+#  false
+#
+[[rules]]
+groups = ["boolean_expression_simplify"]
+name = "simplify_false_and_something"
+query = """
+(
+    (binary_expression
+        left: [(false) (parenthesized_expression (false))]
+        operator : "&&"
+        right : (_) @rhs
+    ) @binary_expression
+)
+"""
+replace = "false"
+replace_node = "binary_expression"
+
+# Before :
+#  abc() && false
+# After :
+#  false
+#
+[[rules]]
+name = "simplify_something_and_false"
+groups = ["boolean_expression_simplify"]
+query = """
+(
+    (binary_expression
+        left : ([
+                (identifier)
+                (true)
+                (false)
+                (selector_expression)
+            ]) @lhs
+        operator : "&&"
+        right: [(false) (parenthesized_expression (false))]
+    ) @binary_expression
+)
+"""
+replace = "false"
+replace_node = "binary_expression"
+
+# Before :
+#  something || true
+# After :
+#  true
+#
+# Note that this rule *won't* rewrite when @lhs is a call.
+[[rules]]
+groups = ["boolean_expression_simplify"]
```

Comment:
We have to think of a strategy to prevent it from simplifying after `<something> || true`

Reply:
@ketkarameya should we deal with this in this PR? or do we open a new issue after this is accepted?

Reply:
i think a follow up would be ok. We can address this alongside the fix for java/kt